### PR TITLE
feat: support <details> tag for GithubHtmlView (#528)

### PR DESCRIPTION
**Pull request by whitedogg13**
| Question         | Response    |
| ---------------- | ----------- |
| Version?         | v1.4.1      |
| Devices tested?  | iPhone X |
| Bug fix?         | no      |
| New feature?     | yes      |
| Includes tests?  | no      |
| All Tests pass?  | yes      |
| Related ticket?  | #528   |

---

## Screenshots

Before (not support details tag)
image

After 1 (detail not expanded)
image

After 2 (detail expanded)
image

## Description

Support `<details>` tag in `GithubHtmlView`, as mentioned in #528 .

First try to find `<summary>` tag inside `<details>`, then render a toggle-able view if found, otherwise do a rollback rendering. Also try to prettify the text in `<summary>` if we got only one text child inside it.

Thanks!

### CODE REVIEW

**Comment by lex111** [comment]
@whitedogg13 it's awesome! :clap: 

What do you think can write a small test for this component?

**Comment by whitedogg13** [comment]
@lex111 Sure, love to do so. I will update it later.

**Comment by whitedogg13** [comment]
Hi @chinesedfan , thanks for the review!

I made some enhancement over your first and third suggestions:
* enhance toggle-view.component.js to replace WithToggle
* add different edge cases to tests

It's always good to test and reuse code.

However, about the handling for `<summary/>` tag, I have checked #609, some points listed:
* it adds a extra `<hidden/>` tag in order to parse `<summary/>`, since `<hidden/>` is not a valid HTML tag, so more or less like a hack to me. 
* it renders `<summary/>` with `<View>{defaultRenderer(node.children, parent)}</View>`, however, this basically means we don't even need to handle this case since it simply do the default behavior.

Because of above reasons, I think current version makes more sense to me.

Thanks!

**Comment by whitedogg13** [comment]
@chinesedfan Thanks for suggestions, I will update it later :)

**Comment by whitedogg13** [comment]
Hi @chinesedfan , I have pushed the changes, thanks!

**Comment by lex111** [comment]
@machour hi, can we merge this PR?

**Comment by machour** [comment]
Hey @lex111, sorry I've been away from computer these days. Have this PR been tested on Android? If it's working there too then you have my blessing :)

**Comment by whitedogg13** [comment]
Hi @lex111 I have just tested this on Nexus 5 using Genymotion Emulator, and it works correctly. For further PR, I will test it in Android as well :)

File: src/components/toggle-view.component.js [hunk 1]
```diff
@@ -28,7 +29,9 @@ export class ToggleView extends Component {
     return (
       <View>
         <TouchableOpacity onPress={() => this._toggle()}>
-          {this.props.TouchableView}
+          {this.props.renderTouchable
+            ? this.props.renderTouchable(this.state.collapsed)
+            : this.props.TouchableView}
```
Comment (housseindjirdeh) [inline src/components/toggle-view.component.js]: 👏
